core-libs/setup/ssr/engine-decorator/ng-express-engine-decorator.ts

```diff
@@ -108,15 +108,17 @@ function getRequestUrl(req: Request): string {
 function getRequestOrigin(req: Request): string {
   // If express is resolving and trusting X-Forwarded-Host, we want to take it
   // into an account to properly generate request origin.
-  if (req.get('X-Forwarded-Host')?.includes(req.hostname)) {
-    let port = ':' + req.get('X-Forwarded-Port');
-    if (
-      (port === ':80' && req.protocol === 'http') ||
-      (port === ':443' && req.protocol === 'https')
-    ) {
-      port = '';
+  const trustProxyFn = req.app.get('trust proxy fn');
+  let forwardedHost = req.get('X-Forwarded-Host');
+  if (forwardedHost && trustProxyFn(req.connection.remoteAddress, 0)) {
```

Comment:
I'm having hard time finding express js docs that help me decipher `trustProxyFn(req.connection.remoteAddress, 0)`. Can we add a comment what and why we check here?

Btw. perhaps we might also want to drop a link here to the express js implementation
https://github.com/expressjs/express/blob/b93ffd4bdc09c3af925eed80c28bd37f63bb3cfc/lib/request.js#L427-L431